src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,38 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_VALUE = 10;
+    private final Object[] keys = new Object[MAX_VALUE];
+    private final Object[] values = new Object[MAX_VALUE];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == null && key == null || keys[i] != null && keys[i].equals(key)) {
+                values[i] = value;
+                return;
+            }
+        }
+        if (size < MAX_VALUE) {
```

When the storage is full (size >= MAX_VALUE) the current code silently ignores an insertion of a distinct key. The task/checklist expects a defined, verifiable behavior for full capacity (either reject/return/throw). Make this behavior explicit (document it or change the API to return boolean / throw) so tests can rely on it.